node/src/components/consensus/era_supervisor/era.rs

```diff
@@ -85,78 +80,62 @@ impl<I> Era<I> {
             consensus,
             start_time,
             start_height,
-            candidates: Vec::new(),
+            validation_states: HashMap::new(),
             newly_slashed,
             slashed,
             accusations: HashSet::new(),
             validators,
         }
     }
 
-    /// Adds a new candidate block, together with the accusations for which we don't have evidence
-    /// yet.
-    pub(crate) fn add_candidate(
+    /// Adds a new block, together with the accusations for which we don't have evidence yet.
+    pub(crate) fn add_block(
         &mut self,
-        candidate: CandidateBlock,
+        proposed_block: ProposedBlock<ClContext>,
         missing_evidence: Vec<PublicKey>,
     ) {
-        self.candidates
-            .push(PendingCandidate::new(candidate, missing_evidence));
+        self.validation_states
+            .insert(proposed_block, ValidationState::new(missing_evidence));
     }
 
-    /// Marks the dependencies of candidate blocks on evidence against validator `pub_key` as
-    /// resolved and returns all candidates that have no missing dependencies left.
-    pub(crate) fn resolve_evidence(&mut self, pub_key: &PublicKey) -> Vec<CandidateBlock> {
-        for pc in &mut self.candidates {
+    /// Marks the dependencies of blocks on evidence against validator `pub_key` as resolved and
+    /// returns all valid blocks that have no missing dependencies left.
+    pub(crate) fn resolve_evidence(
+        &mut self,
+        pub_key: &PublicKey,
+    ) -> Vec<ProposedBlock<ClContext>> {
+        for pc in self.validation_states.values_mut() {
             pc.missing_evidence.retain(|pk| pk != pub_key);
         }
         self.consensus.mark_faulty(pub_key);
-        self.remove_complete_candidates()
+        let (complete, validation_states): (HashMap<_, _>, HashMap<_, _>) = self
```

Comment:
```suggestion
        let (complete, incomplete): (HashMap<_, _>, HashMap<_, _>) = self
```
Can we change the names to be somewhat clearer what it is?

Reply:
Done in https://github.com/casper-network/casper-node/pull/1290/commits/794ed59f41d5773bdfe4357c8ae592e4f9670aaa.